src/docs_versions_menu/parse_version.py

```diff
@@ -0,0 +1,26 @@
+"""Parse a version string
+
+The ``parse_version`` function provided here is functionally equivalent to
+``packaging.version.parse`` in ``packaging < 22.0``.
+
+We leverage the ``packvers`` package for the desired behavior.
+"""
+import packvers.version
+
+
+class NonVersionFolderName(packvers.version.LegacyVersion):
+    """A "version" that is just an arbitrary folder name"""
+
+    # We rely on the sorting properties of LegacyVersion relative to to a
+    # proper Version, even though arbitrary folder names aren't in any sense
+    # "legacy versions"
+    pass
+
+
+def parse_version(version):
+    """Parse `version` string into either a `Version` or a
+    `NonVersionFolderName` object."""
+    try:
+        return packvers.version.Version(version)
+    except packvers.version.InvalidVersion:
+        return NonVersionFolderName(version)
```

Comment:
Why not use `packvers.version.parse` (it would handle LegacyVersion for you; [ref](https://github.com/nexB/packvers/blob/main2/docs/version.rst#reference)) ?
Otherwise with this `try`/`except` you could continue using `packaging.version.parse` instead of `packvers`.

Reply:
I was trying to replace `LegacyVersion` with the `NonVersionFolderName` subclass of `LegacyVersion` that I'm defining to clarify that it's not really a "legacy version". You're right that I could use `packaging` or `packvers` interchangeably here, but I still need `packvers.version.LegacyVersion` to subclass from. The `packaging` package completely removed `LegacyVersion`, right? So, I still need the `packvers` dependency, and at that point I might as well use `packvers` for `parse`/`Version` instead of `packaging`.